Her-Tech-Connect/Her-Tech-Connect/NewEventScreen.swift

```diff
@@ -48,7 +62,63 @@ struct EventDetail: View {
     }
 }
 
+struct AddEventView: View {
+    var body: some View {
+        //code for adding new event.
+        //Text("Nothing here yet?")
+        
+        VStack{
+        TextField(" Event Name", text: $nameField)
+            .background(Color(.white))
+        TextField(" Event Address", text: $addressField)
+                .background(Color(.white))
+        DatePicker("date:", selection: $dateField, displayedComponents: [.date])
+            .labelsHidden()
+        DatePicker("time:", selection: $timeField, displayedComponents: [.hourAndMinute])
+                .labelsHidden()
+        }
+        Spacer()
+        Button(action: {
+            let newLocation = MKPointAnnotation()
+            print("button pressed")
+            if (addressField == ""){
+                return
+            }
+            let formatter1 = DateFormatter()
+            formatter1.dateStyle = .short
+            let formatter2 = DateFormatter()
+            formatter2.timeStyle = .short
 
+            let geocoder = CLGeocoder()
+            geocoder.geocodeAddressString(addressField) {
+                placemarks, error in
+                let placemark = placemarks?.first
+                let lat = placemark?.location?.coordinate.latitude
+                let long = placemark?.location?.coordinate.longitude
+                newLocation.coordinate = CLLocationCoordinate2D(latitude:lat!, longitude: long!)
+                self.locations.append(newLocation)
```

Comment:
Good work with the latitude and longitude. I wonder if you might just store the address and [then compute its location from MapKit](https://mhorga.org/2015/08/14/geocoding-in-ios.html)